Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -177,14 +178,25 @@ class HomeFragmentViewModel @AssistedInject constructor(
         val stateRAT = testRAT.toSubmissionState(timeStamper.nowUTC, coronaTestParameters)
         val pcrIdentifier = testPCR?.identifier ?: ""
         val ratIdentifier = testRAT?.identifier ?: ""
+
+        val positiveCoronaTests = listOfNotNull(testRAT, testPCR).filter { it.isPositive && it.isViewed }
+
         mutableListOf<HomeItem>().apply {
             when {
-                statePCR is SubmissionStatePCR.TestPositive || statePCR is SubmissionStatePCR.SubmissionDone -> {
-                    // Don't show risk card
-                }
-                stateRAT is SubmissionStateRAT.TestPositive || stateRAT is SubmissionStateRAT.SubmissionDone -> {
-                    // Don't show risk card
+                // High risk is always visible regardless of test result
+                tracingItem is IncreasedRiskCard.Item -> add(tracingItem)
+
+                // No high risk card -> check corona test age against config duration
+                positiveCoronaTests.isNotEmpty() -> {
```

Comment:
What about the failed card?

Reply:
At this point risk cards ,but high will be added .Failed Risk, Low Risk Card will be added here if we have all positive are older than the configuration